### PR TITLE
[Merged by Bors] - chore(Algebra/Group/Aut): Do not import `MonoidWithZero`

### PR DESCRIPTION
**Pull request by YaelDillies**
Move what needs it to a new file `Algebra.GroupWithZero.Action.Basic` which will eventually contain more content coming from `Algebra.SMulWithZero`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
Regarding the import changes to `GroupTheory.GroupAction.Group`: This file will soon stop existing, so don't worry too much

File: Mathlib/Algebra/GroupWithZero/Action/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+/-- Scalar multiplication as a monoid homomorphism with zero. -/
+@[simps]
+def smulMonoidWithZeroHom {α β : Type*} [MonoidWithZero α] [MulZeroOneClass β]
+    [MulActionWithZero α β] [IsScalarTower α β β] [SMulCommClass α β β] : α × β →*₀ β :=
+  { smulMonoidHom with map_zero' := smul_zero _ }
+
+section MulDistribMulAction
+
+variable [Group α] [Monoid β] [MulDistribMulAction α β]
```
Comment (grunweg) [inline Mathlib/Algebra/GroupWithZero/Action/Basic.lean:111]: Can you file a follow-up PR to rename these variables, say \alpha to a Group and \beta to M?

**Review by grunweg — COMMENTED** [review]
The code motion part looks good. This PR needs a shake; I didn't check the new module docstring very carefully.

For the next time: this PR was more difficult to review since you mixed in related changes (theorem -> lemma or renaming variables) with code motion. Can you at least split these into separate commits?

File: Mathlib/Algebra/GroupWithZero/Action/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+This file should avoid depending on other parts of `GroupTheory`, to avoid import cycles.
+More sophisticated lemmas belong in `GroupTheory.GroupAction`.
+
+## Tags
+
+group action
+-/
+
+-- TODO:
+-- assert_not_exists Ring
```
Comment (j-loreaux) [inline Mathlib/Algebra/GroupWithZero/Action/Basic.lean:50]: Should this disappear, or be replaced with a different `assert_not_exists`?

File: Mathlib/Algebra/GroupWithZero/Action/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+section DistribMulAction
+variable [Group G] [Monoid M] [AddMonoid A] [DistribMulAction M A]
+
+-- Porting note: this probably is no longer relevant.
+/-! Since Lean 3 does not have definitional eta for structures, we have to make sure
+that the definition of `DistribMulAction.toDistribSMul` was done correctly,
+and the two paths from `DistribMulAction` to `SMul` are indeed definitionally equal. -/
+example :
+    (DistribMulAction.toMulAction.toSMul : SMul M A) = DistribMulAction.toDistribSMul.toSMul := rfl
```
Comment (j-loreaux) [inline Mathlib/Algebra/GroupWithZero/Action/Basic.lean:78]: Can we just remove this? Or at least change the comment beforehand (and remove the porting note).

**Review by j-loreaux — APPROVED** [review]
Mostly LGTM.

File: Mathlib/Algebra/GroupWithZero/Action/Basic.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Algebra/GroupWithZero/Action/Basic.lean:50]: A future PR will actually enforce it, so I think it's perfectly fine as it is now. It shows the intent

File: Mathlib/Algebra/GroupWithZero/Action/Basic.lean (hunk 3 above)
Comment (YaelDillies) [reply Mathlib/Algebra/GroupWithZero/Action/Basic.lean:78]: Yeah sure. Will remove when I'm at a computer, or feel to do it yourself

File: Mathlib/Algebra/GroupWithZero/Action/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/GroupWithZero/Action/Basic.lean:111]: I have renamed the variables in this PR

**Comment by j-loreaux** [comment]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**